phantom-dsl/src/main/scala/com/websudos/phantom/builder/query/prepared/PreparedUpdateQuery.scala

```diff
@@ -163,16 +173,32 @@ sealed class AssignmentsQuery[
     new ConditionalQuery(table, init, usingPart, wherePart, setPart, casPart append query)
   }
 
-  def consistencyLevel_=(level: ConsistencyLevel)(implicit ev: Status =:= Unspecified): AssignmentsQuery[Table, Record, Limit, Order, Specified, Chain] = {
-    new AssignmentsQuery(
-      table,
-      init,
-      usingPart append QueryBuilder.consistencyLevel(level.toString),
-      wherePart,
-      setPart,
-      casPart
-    )
+  def consistencyLevel_=(level: ConsistencyLevel)(implicit ev: Status =:= Unspecified, session: Session): AssignmentsQuery[Table, Record, Limit, Order, Specified, Chain] = {
+    val protocol = session.getCluster.getConfiguration.getProtocolOptions.getProtocolVersionEnum
+
+    if (protocol.compareTo(ProtocolVersion.V2) == 1) {
+      new AssignmentsQuery(
+        table,
+        init,
+        usingPart,
+        wherePart,
+        setPart,
+        casPart,
+        level
+      )
+    } else {
+      new AssignmentsQuery(
+        table,
+        init,
+        usingPart append QueryBuilder.consistencyLevel(level.toString),
```

Comment:
The level can be null at this point. Is `consistencyLevel` setter mandatory? If not why not to make the level field an `Option[ConsistencyLevel]` and call the append method only when the level is not `None`?